clients/drcachesim/scheduler/scheduler.cpp

```diff
@@ -3000,11 +2999,14 @@ scheduler_tmpl_t<RecordType, ReaderType>::pick_next_input_as_previously(
     // enforce finer-grained timing replay.
     if (options_.deps == DEPENDENCY_TIMESTAMPS) {
         for (int i = 0; i < static_cast<output_ordinal_t>(outputs_.size()); ++i) {
-            if (i != output &&
-                outputs_[i].record_index + 1 <
-                    static_cast<int>(outputs_[i].record.size()) &&
-                segment.timestamp >
-                    outputs_[i].record[outputs_[i].record_index + 1].timestamp) {
+            if (i == output)
+                continue;
+            // Do an atomic load once and use it to de-reference if it's not at the end.
+            // If the target advances to the end concurrently with us that's fine:
+            // an extra wait will just come back here and then continue.
```

Comment:
How about starting sentence #2 with the magic words ("this is safe") to make it super clear, e.g. "This is safe because if the target advances to the end concurrently it will only cause an extra wait that will come back here and then continue."

Reply:
Done.